components/script/dom/html/htmlbuttonelement.rs

```diff
@@ -358,13 +358,21 @@ impl Activatable for HTMLButtonElement {
         !self.upcast::<Element>().disabled_state()
     }
 
-    // https://html.spec.whatwg.org/multipage/#run-post-click-activation-steps
-    fn activation_behavior(&self, _event: &Event, _target: &EventTarget, can_gc: CanGc) {
+    // https://html.spec.whatwg.org/multipage/#the-button-element:activation-behaviour
+    fn activation_behavior(&self, _event: &Event, target: &EventTarget, can_gc: CanGc) {
         let ty = self.button_type.get();
         match ty {
-            // https://html.spec.whatwg.org/multipage/#attr-button-type-submit-state
+            // https://html.spec.whatwg.org/multipage/#the-button-element:attr-button-type-submit-state
             ButtonType::Submit => {
-                // TODO: is document owner fully active?
+                // Step 2. If element's node document is not fully active, then return.
+                if !target
+                    .downcast::<Node>()
+                    .unwrap()
```

Comment:
If it's possible that the `EventTarget` isn't a node, you should gracefully handle this failure and not panic here.

```
if !target.downcast::<Node>().is_none_or(|node| node.owner_document().is_fully_active()) {
    return;
}
```